src/arch/cuda/hybridAPI/hapi_impl.cpp

```diff
@@ -231,8 +312,97 @@ void hapiMapping(char** argv) {
       CmiAbort("Unsupported mapping type!");
   }
 
-  // Set device for each PE
+  // Set device and store PE-device mapping
   hapiCheck(cudaSetDevice(CpvAccess(my_device)));
+#if CMK_SMP
+  CmiLock(CsvAccess(gpu_manager).device_mapping_lock);
+#endif
+  CsvAccess(gpu_manager).device_map.emplace(CmiMyPe(),
+      &(CsvAccess(gpu_manager).device_managers[CpvAccess(my_device)]));
+#if CMK_SMP
+  CmiUnlock(CsvAccess(gpu_manager).device_mapping_lock);
+#endif
```

Comment:
Here and elsewhere, is there a drawback to adding `GPUManager & my_gpu_manager = CsvAccess(gpu_manager);` near the top of functions and replacing uses of `CsvAccess(gpu_manager)` with this new local variable? Similarly for other globals with multiple uses in a function.

Reply:
Not really, it just seemed like convention to use `CsvAccess` explicitly similar to how it is used elsewhere in the Charm++ source code. I guess it does have the benefit of reminding us that the variable is process-shared, but I am for your suggestion (it was cumbersome to add `CsvAccess` all the time when writing the code).